modules/abstract-cosmos/src/lib/utils.ts

```diff
@@ -646,10 +646,11 @@ export class CosmosUtils implements BaseUtils {
   /**
    * Validates an array of coin amounts.
    * @param {Coin[]} amountArray - The array of coin amounts to validate.
+   * @param {number} transactionType - optional field for transaction type
    */
-  validateAmountData(amountArray: Coin[]): void {
+  validateAmountData(amountArray: Coin[], transactionType?: number): void {
```

Comment:
why type is number ??

Reply:
Unsure of the historical reasons but `TransactionType.ContractCall` is a number, hence using the concrete type

Reply:
changing it to enum to look cleaner, is that what you meant ?

Reply:
yes using TransactionType looks better.